### ISSUE TITLE
PHP Fatal error:  Interface 'Composer\Plugin\Capability\CommandProvider' not found

### ISSUE THREAD

**Issue by springerin** [vaimo/composer-patches#31]
**Pre-requisites**
1. magento 2.1.9
2. module version 4.14.0

**To Reproduce**
1. composer update to 4.16.0
2. bin/magento c:f

**Expected**
```
Flushed cache types:
config
layout
block_html
collections
reflection
db_ddl
eav
customer_notification
config_integration
config_integration_api
full_page
config_webservice
translate
```

**Actual**
```
PHP Fatal error:  Interface 'Composer\Plugin\Capability\CommandProvider' not found in /vendor/vaimo/composer-patches/src/Composer/CommandsProvider.php on line 8
PHP Stack trace:
PHP   1. {main}() /bin/magento:0
PHP   2. Magento\Framework\Console\Cli->__construct($name = *uninitialized*, $version = *uninitialized*) /bin/magento:22
PHP   3. Magento\Framework\App\ProductMetadata->getVersion() /vendor/magento/framework/Console/Cli.php:81
PHP   4. Magento\Framework\App\ProductMetadata->getSystemPackageVersion() /vendor/magento/framework/App/ProductMetadata.php:65
PHP   5. Magento\Framework\Composer\ComposerInformation->getSystemPackages() /vendor/magento/framework/App/ProductMetadata.php:104
PHP   6. Magento\Framework\Composer\ComposerInformation->getLocker() /vendor/magento/framework/Composer/ComposerInformation.php:242
PHP   7. Magento\Framework\Composer\ComposerInformation->getComposer() /vendor/magento/framework/Composer/ComposerInformation.php:375
PHP   8. Magento\Framework\Composer\ComposerFactory->create() /vendor/magento/framework/Composer/ComposerInformation.php:362
PHP   9. Composer\Factory::create($io = *uninitialized*, $config = *uninitialized*, $disablePlugins = *uninitialized*) /vendor/magento/framework/Composer/ComposerFactory.php:47
PHP  10. Composer\Factory->createComposer($io = *uninitialized*, $localConfig = *uninitialized*, $disablePlugins = *uninitialized*, $cwd = *uninitialized*, $fullLoad = *uninitialized*) /vendor/composer/composer/src/Composer/Factory.php:532
PHP  11. Composer\Plugin\PluginManager->loadInstalledPlugins() /vendor/composer/composer/src/Composer/Factory.php:354
PHP  12. Composer\Plugin\PluginManager->loadRepository($repo = *uninitialized*) /vendor/composer/composer/src/Composer/Plugin/PluginManager.php:76
PHP  13. Composer\Plugin\PluginManager->registerPackage($package = *uninitialized*, $failOnMissingClasses = *uninitialized*) /vendor/composer/composer/src/Composer/Plugin/PluginManager.php:240
PHP  14. Composer\Plugin\PluginManager->addPlugin($plugin = *uninitialized*) /vendor/composer/composer/src/Composer/Plugin/PluginManager.php:188
PHP  15. Vaimo\ComposerPatches\Plugin->activate($composer = *uninitialized*, $io = *uninitialized*) /vendor/composer/composer/src/Composer/Plugin/PluginManager.php:215
PHP  16. Vaimo\ComposerPatches\Bootstrap->preloadPluginClasses() /vendor/vaimo/composer-patches/src/Plugin.php:42
PHP  17. Vaimo\ComposerPatches\Package\SourcesPreloader->preload($package = *uninitialized*) /vendor/vaimo/composer-patches/src/Bootstrap.php:122
PHP  18. require_once() /vendor/vaimo/composer-patches/src/Package/SourcesPreloader.php:50
```

**Comment by allanpaiste** [comment]
Yeah this one is pretty tricky, because the module introduces new Composer commands, but the interfaces for those did not become available til Composer 1.1 AND M2.1.X has composer locked on some ancient BETA version :(

Will address this by just not providing the commands for that specific context. 

Meanwhile you could get around this problem with aliasing in a bit newer composer (there's no need why any application should use beta version of it anymore):

```
# The 1.1.0 is also quite old, but using it here as it's the closest to what M2.1 demands 
# and the first that has the missing interface
composer require composer/composer:'1.1.0 as 1.0.0-beta1'
```

**Comment by allanpaiste** [comment]
Fixed in 3.47.1 & 4.16.1
